README.md

```diff
@@ -1,5 +1,245 @@
+
+## API documentation
+### Endpoints available
+<table class="responsive fixed orange">
+<colgroup><col width="214px"><col></colgroup>
+<tr><th><h3 class="add-link">Endpoint</h3></th>
+<th><h3 class="add-link">Details</h3></th></tr>
+<tr>
+<td>
+<code>chat/</code>
+</td>
+<td>
+This is the <b>non-streaming</b> POST method route. Use this to send the chat message and the history of the conversation to the Gemini model. The complete response generated by the model to the posted message will be returned in the API's response.  
+<br />
+<br />
+<code>POST</code> <code><b>chat/</b></code>
+<h4>Parameters</h4>
+<table class="responsive fixed orange">
+<tr><td><h5>Name</h5></td><td><h5>Type</h5></td><td><h5>Data type</h5></td><td><h5>Description</h5></td></tr>
+<tr><td>chat</td><td>required</td><td>string</td><td>Latest chat message from user</td></tr>
+<tr><td>history</td><td>optional</td><td>array</td><td>Current chat history between user and Gemini model</td></tr>
+</table>
+<h4>Response</h4>
+<table class="responsive fixed orange">
+<tr><td><h5>HTTTP code </h5></td><td><h5>Content-Type</h5></td><td><h5>Response</h5></td></tr>
+<tr><td>200</td><td>application/json</td><td>{"text": string}</td></tr>
+</table>
+</td>
+</tr>
+<tr>
+<td>
+<code>stream/</code>
+</td>
+<td>
+This is the <b>streaming</b> POST method route. Use this to send the chat message and the history of the conversation to the Gemini model. The response generated by the model will be streamed to handle partial results.  
+<br />
+<br />
+<code>POST</code> <code><b>stream/</b></code>
+<h4>Parameters</h4>
+<table class="responsive fixed orange">
+<tr><td><h5>Name</h5></td><td><h5>Type</h5></td><td><h5>Data type</h5></td><td><h5>Description</h5></td></tr>
+<tr><td>chat</td><td>required</td><td>string</td><td>Latest chat message from user</td></tr>
+<tr><td>history</td><td>optional</td><td>array</td><td>Current chat history between user and Gemini model</td></tr>
+</table>
+<h4>Response</h4>
+<table class="responsive fixed orange">
+<tr><td><h5>HTTTP code </h5></td><td><h5>Content-Type</h5></td><td><h5>Response</h5></td></tr>
+<tr><td>200</td><td>application/json</td><td>string</td></tr>
+</table>
+</td>
+</tr>
+</table>
+
+## Installation
+Click [here](#windows) to skip to installation on Windows.
+
+For Linux/macOS a setup bash script is available for easy installation. If you prefer installing manually you can skip to [next section](#install-manuallylinuxmacoswindows).
+### Using bash script(Linux/macOS)
+#### Make the script executable
+```
+chmod +x setup.sh
+```
+#### Installation options
+
+You can choose to install the required packages for
+- Frontend(React) plus a backend server of your choice(Flask or Node.js).
+- Frontend(React) and both backend servers(Flask, and Node.js).
+
+You can specify your choice using the `BACKEND` variable while running the script.
+
+Install React along with Node.js backend.
+```
+BACKEND=nodejs ./setup.sh
```

Comment:
Does this need the `source` command too? Just noticed the other two install commands have it.

Reply:
No. Source is only needed for Python venv setup. When BACKEND=nodejs, it is not needed.